pkg/binlog/filename_test.go

```diff
@@ -23,7 +23,101 @@ var _ = Suite(&testFilenameSuite{})
 type testFilenameSuite struct {
 }
 
-func (t *testFilenameSuite) TestVerifyBinlogFilename(c *C) {
+func (t *testFilenameSuite) TestFilenameCmp(c *C) {
+	f1 := Filename{
+		BaseName: "mysql-bin",
+		Seq:      "000001",
+	}
+	f2 := Filename{
+		BaseName: "mysql-bin",
+		Seq:      "000002",
+	}
+	f3 := Filename{
+		BaseName: "mysql-bin",
+		Seq:      "000001", // == f1
+	}
+	f4 := Filename{
+		BaseName: "bin-mysq", // diff BaseName
+		Seq:      "000001",
+	}
+
+	c.Assert(f1.LessThan(f2), IsTrue)
+	c.Assert(f1.GreaterOrEqualThan(f2), IsFalse)
+	c.Assert(f1.GreaterThan(f2), IsFalse)
+
+	c.Assert(f2.LessThan(f1), IsFalse)
+	c.Assert(f2.GreaterOrEqualThan(f1), IsTrue)
+	c.Assert(f2.GreaterThan(f1), IsTrue)
+
+	c.Assert(f1.LessThan(f3), IsFalse)
+	c.Assert(f1.GreaterOrEqualThan(f3), IsTrue)
+	c.Assert(f1.GreaterThan(f3), IsFalse)
+
+	c.Assert(f1.LessThan(f4), IsFalse)
+	c.Assert(f1.GreaterOrEqualThan(f4), IsFalse)
+	c.Assert(f1.GreaterThan(f4), IsFalse)
+}
+
+func (t *testFilenameSuite) TestParseFilename(c *C) {
+	cases := []struct {
+		filenameStr string
+		filename    Filename
+		errMsgReg   string
+	}{
+		{
+			// valid
+			filenameStr: "mysql-bin.666666",
+			filename:    Filename{"mysql-bin", "666666"},
+		},
+		{
+			// empty filename
+			filenameStr: "",
+			errMsgReg:   ".*invalid binlog filename.*",
+		},
+		{
+			// negative seq number
+			filenameStr: "mysql-bin.-666666",
+			errMsgReg:   ".*invalid binlog filename.*",
+		},
+		{
+			// zero seq number
+			filenameStr: "mysql-bin.000000",
+			errMsgReg:   ".*invalid binlog filename.*",
+		},
+		{
+			// too many separators
+			filenameStr: "mysql.bin.666666",
+			errMsgReg:   ".*invalid binlog filename.*",
+		},
+		{
+			// too less separators
+			filenameStr: "mysql-bin",
+			errMsgReg:   ".*invalid binlog filename.*",
+		},
+		{
+			// invalid seq number
+			filenameStr: "mysql-bin.666abc",
+			errMsgReg:   ".*invalid binlog filename.*",
+		},
+		{
+			// invalid seq number
+			filenameStr: "mysql-bin.def666",
+			errMsgReg:   ".*invalid binlog filename.*",
+		},
```

Comment:
how about add a case `mysql.bin`?

Reply:
done in https://github.com/pingcap/dm/pull/140/commits/cf17487d1339271972b9846d950214c23e88e13a